sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -387,7 +400,27 @@ public void testGroupBy() {
       },
       Encoders.STRING());
 
-    Assert.assertEquals(asSet("1a#2", "3foobar#6", "5#10"), toSet(cogrouped.collectAsList()));
+    Assert.assertEquals(asSet("1a#2", "3foobar#67", "5#10"), toSet(cogrouped.collectAsList()));
+
+    Dataset<String> cogroupSorted = grouped.cogroupSorted(
+      grouped2,
+      JavaConverters.iterableAsScalaIterable(Collections.singletonList(ds.col("value"))).toSeq(),
+      JavaConverters.iterableAsScalaIterable(Collections.singletonList(ds2.col("value").desc())).toSeq(),
```

Comment:
If we can't make `Column*` java friendly, `Array[Column]` seems better

Reply:
What about the Java `Iterator`s given to the Scala function? I don't think this is solid (by itself, and compared to existing `flatMapGroups` and `cogroup`).

Reply:
good point. Even if the lambda problem is resolved, but the collection type is still incompatible between Java and Scala. Let's still create 2 overloads. Sorry for the back and forth!

Reply:
I'll revert the API then. No worries, it is well worth checking for the best API, once released it cannot be changed.